lib/bootscale/utils.rb

```diff
@@ -0,0 +1,30 @@
+module Bootscale
+  module Utils
+    # Write to a file atomically. Useful for situations where you
+    # don't want other processes or threads to see half-written files.
+    #
+    #   Utils.atomic_write('important.file') do |file|
+    #     file.write('hello')
+    #   end
+    #
+    # Returns nothing.
+    def self.atomic_write(filename)
```

Comment:
We can also borrow them from Sprockets 😀  https://github.com/rails/sprockets/blob/eecb2d47ded089e464883389c9efc325bc1177a9/test/test_path_utils.rb#L288